### PR TITLE
add linters action and publish action

### PR DESCRIPTION
**Pull request by kasif-adnan**
# Description

Describe what this change achieves

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Review by sanjain-progress — DISMISSED** [review]
@kasif-adnan Few yamlLint warning are their `missing document start "---"`. 
append `---` starting of the files.

**Review by ashiqueps — DISMISSED** [review]

File: .github/workflows/unit.yml [hunk 1]
```diff
@@ -11,12 +12,12 @@ jobs:
...
+    name: Unit Test with Ruby ${{ matrix.ruby }}
     steps:
       - uses: actions/checkout@v2
       - uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
-      - run: bundle exec rake
\ No newline at end of file
+      - run: bundle exec rake spec
```
Comment (damacus) [inline .github/workflows/unit.yml:23]: Should we move the Unit tests to run after the linters?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+### Description
+
```
Comment (Xorima) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: We should be using org wide templates not repo specific templates

Comment (kasif-adnan) [reply .github/PULL_REQUEST_TEMPLATE.md]: removed it

**Review by damacus — APPROVED** [review]

**Pull request merged**